### ISSUE TITLE
V2: Disabling the client code generation

### ISSUE THREAD

**Issue by nikolaas** [goadesign/goa#1892]
Hello!

Is there a way to turn off the client code generation by the `goa gen` command in Goa V2?

Thanks

**Comment by raphael** [comment]
No there isn't at the moment. I wanted to keep the tool dead simple and only add commands and flags when there is a clear and strong need for them (a la `go` command line).

**Comment by xlab** [comment]
Hi @raphael,

I was wondering if this is a possible flow now with `Meta("swagger:generate", "false")`?
I tried `Meta("cli:generate", "false")` but for obvious reasons it doesn't exist for now :)

**Comment by raphael** [comment]
Yes that would work - it would fit with the intent behind `Meta` (modifying the behavior of code generators). I would merge a PR that added this behavior (I guess the PR should probably also add a `server:generate` metadata to satisfy the principle of least surprise :) ).

**Comment by xlab** [comment]
@raphael sorry, can you point out to the example of disabling CLI generation now?

I occasionally get these errors:

```
api/gen/http/rest_api/client/cli.go:220:6: err declared but not used
```

But I don't use CLI and aways try to manually remove `cli.go` files because of viper/cobra dependency.

**Comment by raphael** [comment]
@xlab just making sure I understand: the code that Goa (which version?) generates is sometimes incorrect? If that's true it would really help if you could test whether the issue still exists with the latest version. It it does then please open a separate issue with a small complete repro so we can fix the problem (Goa should never generate code that does not compile obviously :) ).

In terms of disabling CLI generation this is not something that exists at this point. Honestly I'm not convinced of the usefulness of such a flag, there really shouldn't be any real impact whether the code ends up being used or not. But if there is a strong desire to enable that functionality then I'd merge a PR that added the ability to disable that generation via metadata as described above.

**Comment by tchssk** [comment]
FYI: I created a plugin to disable a part of the code generator.

https://github.com/tchssk/goaplugins/tree/master/codegendisabler

**Comment by xlab** [comment]
@tchssk very cool! I'll use it

@raphael I'll find some time to provide examples. Our `cli.go` is being generated broken.
